packages/form-js-viewer/src/util/index.js

```diff
@@ -92,11 +94,21 @@ export function getSchemaVariables(schema) {
       variables = [ ...variables, key ];
     }
 
-    if (valuesKey && !variables.includes(valuesKey)) {
+    if (valuesKey) {
       variables = [ ...variables, valuesKey ];
     }
 
-    return variables;
+    if (condition) {
```

Comment:
In the future, we would need to specify any other property here that could be an expression or unary test, right?

Reply:
Yeah or have a separate `if` with logic to handle that property.